livekit-ffi/src/cabi.rs

```diff
@@ -15,13 +15,23 @@ pub type FfiCallbackFn = unsafe extern "C" fn(*const u8, usize);
 ///
 /// The foreign language must only provide valid pointers
 #[no_mangle]
-pub unsafe extern "C" fn livekit_ffi_initialize(cb: FfiCallbackFn, capture_logs: bool) {
+pub unsafe extern "C" fn livekit_ffi_initialize(
+    cb: FfiCallbackFn,
+    capture_logs: bool,
+    sdk: Option<&'static [u8; 16]>,
+    sdk_version: Option<&'static [u8; 16]>,
+) {
+    let sdk = std::str::from_utf8(sdk.unwrap()).unwrap().trim_end_matches('\0');
+    let sdk_version = std::str::from_utf8(sdk_version.unwrap()).unwrap().trim_end_matches('\0');
```

Comment:
i got warnings when using str or String here that they aren't FFI-safe so I used [u8; 16] which meant i also needed to handle nulls below so I made sdk and sdk version optional. I'm going to go ahead and do the python and maybe the unity implementation before I merge this, since I believe this bit here is used in those but not node, so its not tested yet

Reply:
You should use a standard "C" null-terminated string

Reply:
```
use std::os::raw::c_char;

sdk: *const c_char
```

Reply:
thanks that worked great paired with `lib::ffi::CStr`